pkgs/collection/lib/src/priority_queue.dart

```diff
@@ -36,6 +36,19 @@ abstract class PriorityQueue<E> {
   factory PriorityQueue([int Function(E, E)? comparison]) =
       HeapPriorityQueue<E>;
 
+  /// Create a new priority queue.
+  ///
+  /// The [comparison] is a [Comparator] used to compare the priority of
+  /// elements. An element that compares as less than another element has
+  /// a higher priority.
+  ///
+  /// If [comparison] is omitted, it defaults to [Comparable.compare]. If this
+  /// is the case, `E` must implement [Comparable], and this is checked at
+  /// runtime for every comparison.
+  factory PriorityQueue.of(
```

Comment:
What's the right way to easily support `Comparable<T>`

`factory PriorityQueue.ofComparable<T extends Comparable<T>>(Iterable<T>)`...something?

Reply:
That's an option, or just use `Comparable.compare` explicitly.